app/javascript/graph_builder.js

```diff
@@ -20,16 +52,38 @@ document.addEventListener('turbolinks:load', function() {
     document.getElementById('graph-spinner').hidden = true
     document.getElementById('graph-holder').hidden = false
     if (run.run.histories.length !== 0) {
+      run.run.segments.forEach(segment => {
+        build_segment_graph(segment, chartOptions)
+      })
       build_run_duration_graph(run.run, chartOptions)
       buildSegmentGraphs(run.run, chartOptions)
       build_reset_graph(run.run, chartOptions)
       build_playtime_graph(run.run, chartOptions)
     }
   }).catch(function(error) {
+    for (const segGraphAlert of document.getElementsByClassName('segment-graph-alert')) {
+      segGraphAlert.textContent = `Error loading graphs: ${error}`
+      segGraphAlert.hidden = false
+    }
     document.getElementById('graph-spinner').style.display = 'none'
-    document.getElementById('graph-alert').textContent = 'Error loading graphs: ' + error
+    document.getElementById('graph-alert').textContent = `Error loading graphs: ${error}`
     document.getElementById('graph-alert').hidden = false
 
     throw error
   })
 })
+
+document.addEventListener('click', event => {
+  const toggler = event.target.closest('.segment-graph-toggler')
+  if (toggler) {
+    const segId = toggler.dataset.segment
+    const row = document.querySelector(`.segment-graph-holder[data-segment="${segId}"]`).closest('tr')
+    row.hidden = !row.hidden
+    if (!row.hidden) {
+      const segCharts = Highcharts.charts.filter(chart => {
+        return chart.renderTo.id === `segment-graph-holder-${segId}`
+      })
+      segCharts.forEach(chart => { chart.reflow() })
+    }
+  }
```

Comment:
Not sure if the JS linting community agrees but unless we know otherwise, let's stick to the Ruby/Rubocop rule of preferring guard clauses
```javascript
if (!toggler) {
  return
}
// the rest
```
over
```javascript
if (toggler) {
  // the rest
}
```